backend/benefit/applications/tests/test_applications_api.py

```diff
@@ -890,9 +890,12 @@ def test_application_status_change_as_applicant(
         (ApplicationStatus.HANDLING, ApplicationStatus.ACCEPTED, 200),
         (ApplicationStatus.HANDLING, ApplicationStatus.REJECTED, 200),
         (ApplicationStatus.HANDLING, ApplicationStatus.CANCELLED, 200),
+        (ApplicationStatus.ACCEPTED, ApplicationStatus.HANDLING, 200),
+        (ApplicationStatus.REJECTED, ApplicationStatus.HANDLING, 200),
         (ApplicationStatus.RECEIVED, ApplicationStatus.DRAFT, 400),
         (ApplicationStatus.ACCEPTED, ApplicationStatus.RECEIVED, 400),
         (ApplicationStatus.CANCELLED, ApplicationStatus.ACCEPTED, 400),
+        (ApplicationStatus.CANCELLED, ApplicationStatus.HANDLING, 400),
         (ApplicationStatus.REJECTED, ApplicationStatus.DRAFT, 400),
     ],
```

Comment:
How about adding the rest of the cases going back to HANDLING state:

```python
(ApplicationStatus.DRAFT, ApplicationStatus.HANDLING, <status code>),
(ApplicationStatus.RECEIVED, ApplicationStatus.HANDLING, <status code>),
(ApplicationStatus.HANDLING, ApplicationStatus.HANDLING, <status code>),
```

Reply:
The allowed state transitions are defined in the StatusTransitionValidator, I'm only testing a subset of state transitions to ensure the state machine logic works